src/azure-cli/azure/cli/command_modules/acr/agentpool.py

```diff
@@ -0,0 +1,129 @@
+        cmd.cli_ctx, registry_name, resource_group_name)
+
+    AgentPool = cmd.get_models('AgentPool')
+
+    agentpool_create_parameters = AgentPool(
+        location=registry.location,
+        count=count,
+        tier=tier.upper(),
+        os=os_type,
+        virtual_network_subnet_resource_id=subnet_id
+    )
+
+    try:
+        response = client.create(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name,
+                                 agent_pool=agentpool_create_parameters)
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
+        return response
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_update(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         resource_group_name=None,
+                         no_wait=False,
+                         count=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        response = client.update(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name,
+                                 count=count)
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
+        return response
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_agentpool_delete(cmd,
+                         client,
+                         agent_pool_name,
+                         registry_name,
+                         no_wait=False,
+                         resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    try:
+        response = client.delete(resource_group_name=resource_group_name,
+                                 registry_name=registry_name,
+                                 agent_pool_name=agent_pool_name)
+
+        if no_wait:
+            return client.get(resource_group_name, registry_name, agent_pool_name)
+
+        #Since agent pool is a tracked resource in arm, arm also pings the async deletion api at the
+        #same time to get the status. If arm gets the 200 status first and knows that the resource is deleted,
+        #it marks the resource as deleted and stop routing further requests to the resource including the
+        #async deletion status api. Hence arm will directly return 404. Consider this as successful delete.
+        from ._agentpool_polling import delete_agentpool_with_polling
+        delete_agentpool_with_polling(cmd, client, agent_pool_name, registry_name, resource_group_name)
+        return response
```

Comment:
in this case, can you directly return the new agent_poller from `delete_agentpool_with_polling`

Reply:
Yeah, I think you can just return `delete_agentpool_with_polling`

Reply:
good catch. 
This was for the original version of the command. Behavior was a little bit different